tests/kernel/blktests.pm

```diff
@@ -0,0 +1,78 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018-2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: Block device layer tests
+# Maintainer: Michael Moese <mmoese@suse.de>, Sebastian Chlad <schlad@suse.de>
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+
+#use utils 'zypper_call';
+#use power_action_utils 'power_action';
+
+sub run {
+    my $self = shift;
+    $self->select_serial_terminal;
+
+    #below variable exposes blktests options to the openQA testsuite
+    #definition, so that it allows flexible ways of re-runing the tests
+    my $quick   = get_required_var('blk_QUICK');
+    my $exclude = get_required_var('blk_EXCLUDE');
+    my $config  = get_required_var('blk_CONFIG');
+    my $tests   = get_required_var('blk_TESTS');
+    my $device  = get_required_var('blk_DEVICE_ONLY');
+
+    script_run('zypper -n in git-core make gcc coreutils awk util-linux fio gcc-c++',       180);
+    script_run('git -c http.sslVerify=false clone https://github.com/osandov/blktests.git', 120);
+
+    script_run('cd ./blktests');
+    script_run('make && make install');
+
+    script_run('./check loop',                 180);
+    script_run("./check --quick=$quick block", 360);
+    script_run("./check --quick=$quick nvme",  180);
+
+    ##TODO: this can grown into stand-alone "runner" for blktests
+    ##  and perhaps could be done the same way as xfstests? to be checked
+    script_run('wget --quiet ' . data_url('kernel/post_process') . ' -O post_process');
+    script_run('chmod +x post_process');
+    script_run('./post_process');
```

Comment:
If you like you can simplify that to

```perl
script_run('curl -s ' . data_url('kernel/post_process') . ' | sh');
```

Reply:
Overall this part should not live long in this .pm module.